src/parser_fill_cmd.c

```diff
@@ -0,0 +1,168 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser_fill_cmd.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/22 14:59:22 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/26 01:15:14 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file parser_fill_cmd.c
+ * @brief Functions to fill the cmd struct.
+ */
+
+#include "mod_parser.h"
+
+/**
+ * @brief Open file and save fd as fd_in.
+ * 
+ * @param lst 		List of token.
+ * @param new 		New cmd to be filled.
+ * @return t_err 	ERR_BAD_FD, SUCCESS
+ */
+t_err	ft_save_infile(t_tkn_list **lst, t_cmd *new)
+{
+	int			fd_in;
+	t_tkn_list	*tmp;
+
+	tmp = *lst;
+	tmp = tmp->next;
+	new->infile = 1;
+	if (new->fd_in > 0)
+		close(new->fd_in);
+	fd_in = open(tmp->content, O_RDONLY);
+/* 	if (fd_in < 3)
```

Comment:
It is commented out due to the bug of infiles not being consistently opened as they should be. If active this would abort the whole programm and hinder the testing. Needs to be addressed after having found the root cause of the infiles not being opened reliably. Furthermore, why should anyone close any fd 0, 1 or 2? We don't in our programm and it just shouldnt happen if you wanna communicate with shell

Reply:
Ah ok!

Why close fd 2? That's like asking why would anyone open two heredocs :D There are some crazy mofos out there :0 
I would just stick to the convention check for -1 as error. I have not (yet?) seen the check for fd < 3

Reply:
those people should not be let near any PCs. But to be foolproof, changed to  -1